lassen/sim.py

```diff
@@ -49,10 +49,16 @@ def alu(inst:Inst, a:Data, b:Data, d:Bit) -> (Data, Bit, Bit, Bit, Bit,
             b = SInt[datawidth](b)
             mula, mulb = a.sext(16), b.sext(16)
             mul = mula * mulb
+            gte_pred = a >= b
+            lte_pred = a <= b
+            abs_pred = a >= 0
             shr = a >> b
         elif signed == Signed.unsigned:
             mula, mulb = a.zext(16), b.zext(16)
             mul = mula * mulb
+            gte_pred = a >= b
+            lte_pred = a <= b
+            abs_pred = a >= 0
```

Comment:
@Kuree, This looks like repeated code which will translate to more RTL. Can you do this calculation outside the if/elif block?

Reply:
I think @leonardt wrote that code but I can try if it works.

Reply:
This duplication is required by magma, and is in fact required specifically so the RTL is "duplicated", but it's not duplicated because it's doing different ops depending on whether it's signed vs unsigned (so one code path has the signed variant, the other has the unsigned variant).

The more general issue is that magma's sequential syntax doesn't support polymorphic operators, so we need separate statements in the branches because one is actually a signed shift versus an unsigned shift.

Compare the original version where `a` and `b` could be signed or unsigned. magma doesn't know which one it is statically, so we would need to do the analysis to figure out that there's multiple possible types and generate the two code paths + a mux. This is feasible but complicated, so for now we just "duplicate" the code.

Reply:
okay, I will merge